app/models/collection.rb

```diff
@@ -6,6 +6,8 @@ class Collection < ApplicationRecord
 
   validates :slug, presence: true, uniqueness: { scope: :user_id }
 
+  scope :non_empty, -> { joins(:articles).distinct }
```

Comment:
Question (possibly out of scope for this change): should this scope `non_empty` be default for collections? Is there still a place where we want to retrieve and display empty collections?

Reply:
yeah I think it would be safe to make it default

Reply:
@djuber tl;dr: `default_scope` can be painful, I almost always prefer explicit scopes.

Not a big fan of `default_scope` in Rails as it has certain subtleties (i.e. it adds a default `order` clause so if you need custom ordering you need to use `reorder` instead of `order`). I also don't like that it "lies", e.g. with a `default_scope` applied something like `Collection.all` won't actually return all collections. This can become problematic when you do things like `user.collections.destroy_all` which would not remove empty collections. To avoid this you would have to remember that (A) there's a default scope defined on `Collection` and (B) that you need to put `unscoped` before the `destroy_all`.